### PR TITLE
Week 7B features

### PR DESCRIPTION
**Pull request by dcousette**
This request is to merge in the features from part two of week 7 assignments:
- Add a decorator for the video show page using draper 
- Refactor the user sign-in into a service 

-DeShawn

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -7,25 +7,13 @@ def new
...
-        redirect_to signin_path
-      else
-        flash.now[:danger] = charge.error_message
-        render :new
-      end
+    if result.successful?
+      flash[:success] = "Thank you for registering with Myflix. Please sign in now."
+      redirect_to signin_path
     else
+      flash[:danger] = result.error_message
```
Comment (brandoncc) [inline app/controllers/users_controller.rb]: Make sure you use `flash.now` when you use `render`. `flash` will survive the next page render as well.

File: app/services/user_sign_up.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
+class UserSignUp
+  attr_reader :error_message
+
+  def initialize(user)
+    @user = user
+  end
+
+  def sign_up(stripe_token, invitation_token)
```
Comment (brandoncc) [inline app/services/user_sign_up.rb:8]: This method signature requires two arguments. Using an options hash does not. Just something to be aware of. You could also set a default value of `nil` for `invitation_token`.

File: app/services/user_sign_up.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      end
+    else
+      @status = :failed
+      @error_message = 'Invalid user information. Please check the errors below.'
+      self
+    end
+  end
+
+  def successful?
+    @status = :success
```
Comment (brandoncc) [inline app/services/user_sign_up.rb]: You definitely have a typo, `=` should be `==`.

File: spec/controllers/users_controller_spec.rb [hunk 4]
```diff
@@ -9,98 +9,30 @@
...
-
-      it 'creates the user' do
-        post :create, user: Fabricate.attributes_for(:user)
-        expect(User.count).to eq(1)
-      end
-
+    context 'successful user sign up' do
       it 'redirects to the sign in page' do
+        result = double(:sign_up_result, successful?: true)
+        UserSignUp.any_instance.should_receive(:sign_up).and_return(result)
```
Comment (brandoncc) [inline spec/controllers/users_controller_spec.rb]: `should` syntax is no longer recommended. The `expect` syntax equivalent to this is

``` ruby
expect_any_instance_of(UserSignUp).to receive(:sign_up).and_return(result)
```

File: app/services/user_sign_up.rb (hunk 2 above)
Comment (ppj) [inline app/services/user_sign_up.rb:8]: I also like keyword (a.k.a named) arguments in cases where there are 2 or more arguments for clarity of signature.
`def sign_up(stripe_token:, invitation_token: nil)`.

**Pull request merged**